feat: Добавлен режим "Один на один" и добавлены новые команды

ISAYwtf opened the pull request:
Добавлены команда отмены игры, команда удаления игрока и изменена команда выхода из комнаты.

F4lkr4m commented on src/routes/room/index.ts:
```diff
@@ -217,23 +350,39 @@ export const handleTeamWin = async (database: Repo, ctx: ActionCtx, team: 'first
 
     const winner1 = await database.userRepo.getUser(id1);
-    const winner2 = await database.userRepo.getUser(id2);
+    console.log(typeof id2, id2);
```
console.log лишний

F4lkr4m commented on src/routes/room/index.ts:
```diff
@@ -61,6 +68,113 @@ export const removePlayer = authGuardMiddleware(async (database: Repo, ctx: Ctx)
+    return;
+  }
+
+  if (!text.length) {
```
Сверху была проверка на text.length, тут по идее должен быть уже обратный случай

F4lkr4m commented on src/routes/room/index.ts:
```diff
@@ -115,12 +229,46 @@ export const showPlayers = async (database: Repo, ctx: Ctx) => {
+    const total = `\nВсего: ${room.players.length}`;
+
+    let helperText = '';
+    if (room.players.length < 2) {
```
Magic Numbers, вроде были там где то константы на это

F4lkr4m commented on src/routes/room/index.ts:
```diff
@@ -115,12 +229,46 @@ export const showPlayers = async (database: Repo, ctx: Ctx) => {
+        helperText,
+    ].join('\n');
+
+    const replyOptions = (room.players.length === 2 || room.players.length === 4)
```
Ну тут тоже соответственно Magic Numbers

F4lkr4m commented on src/routes/room/index.ts:
```diff
@@ -61,6 +68,113 @@ export const removePlayer = authGuardMiddleware(async (database: Repo, ctx: Ctx)
+      }
+    }
+
+    for (const removingUser of usersToRemove) {
```
Такое ощущение, что тут можно подумать насчет параллельного выполнения, это в целом касается всего кода. По идее в конкретно этом случае должен хорошо асинхронный reduce зайти, но это просто на подумать, делать сейчас не нужно

ISAYwtf replied:
Вообще согласен. Если говорить о "подумать на будущее", то я бы над структурой еще подумал бы

ISAYwtf committed:
feat: Добавлен режим "Один на один" и добавлены новые команды

Добавлены команда отмены игры, команда удаления игрока и изменена команда выхода из команды.
